
/*
      Dopo aver modificato il .proto file:
          python -m grpc_tools.protoc -I../../protos --python_out=. --grpc_python_out=. ../../protos/newpassword.proto

      Questo genera
          - newpassword_pb2.py che contiene le classi di richiesta e risposta
          - newpassword_pb2_grpc.py che contiene le classi client e server

*/

syntax = "proto3";

option java_multiple_files = true;
option java_package = "/SDCC/services";
option java_outer_classname = "NewPwProto";
option objc_class_prefix = "NPW";
package newpw;

// The  service definition.
service Password {

  rpc GetNewPass (PwRequest) returns (PwReply) {}
  rpc GetNewNumPass (PwRequest) returns (PwReply) {}
  rpc GetNewLowerPass (PwRequest) returns (PwReply) {}
  rpc GetNewUpperPass (PwRequest) returns (PwReply) {}
  rpc GetNewAlphaNumPass (PwRequest) returns (PwReply) {}

}


// The request message containing the user's name.
message PwRequest {
  string name = 1;
  int32 length = 2;
  bool symbols = 3;
}

message PwRequestW {
  string name = 1;
  int32 length = 2;
  string kword = 3;
}

// The response message containing the new password
message PwReply {
  string message = 1;
  string pw = 2;
}

